src/commonMain/kotlin/com.adamratzman.spotify/models/SpotifyUris.kt

```diff
@@ -178,13 +164,14 @@ typealias PlaylistURI = PlaylistUri
  * from an endpoint
  * */
 @Serializable
-sealed class TrackUri(input: String, type: String) : SpotifyUri(input, type) {
-    @Serializer(forClass = TrackUri::class)
-    companion object : KSerializer<TrackUri> {
-        override val descriptor: SerialDescriptor = StringDescriptor
-        override fun deserialize(decoder: Decoder): TrackUri = TrackUri(decoder.decodeString())
-        override fun serialize(encoder: Encoder, obj: TrackUri) = encoder.encodeString(obj.uri)
+sealed class TrackUri(
+    @Transient private val _input: String = TRANSIENT_EMPTY_STRING,
+    @Transient private val _type: String = TRANSIENT_EMPTY_STRING
+) :
+    SpotifyUri(_input, _type) {
 
+    @Serializer(forClass = TrackUri::class)
+    companion object {
```

Comment:
This doesn't make any sense. You should to implement `KSerializer` on a object with his annotation.